### PR TITLE
rewrite with Promise and  ESlint

### PR DESCRIPTION
**Pull request by zhangweijie-cn**
R= @yorkie

### CODE REVIEW

File: github.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-'use strict'
```
Comment (yorkie) [inline github.js]: Why removing the strict mode from this file?

File: index.js [hunk 2]
```diff
@@ -1,110 +1,139 @@
-'use strict'
+'use strict';
 
 const koa = require('koa');
 const koaBody = require('koa-body');
 const GithubWebhookHandler = require('koa-github-webhook-handler');
 const github = require('./github');
 const bumpVersion = require('semver-increment');
+const Promise = require('bluebird');
```
Comment (yorkie) [inline index.js]: Why use bluebird instead of the core Promise.

File: github.js (hunk 1 above)
Comment (zhangweijie-cn) [inline github.js]: when run eslint get  `error    'use strict' is unnecessary inside of modules`

File: index.js [hunk 3]
```diff
@@ -1,110 +1,139 @@
...
+  const list = url.split('/');
+  return {
+    user: list[3],
+    repo: list[4],
+    branch: list[6],
+  };
+}
+
+function getPullRequests(url) {
+  return new Promise((resolve, reject) => {
```
Comment (yorkie) [inline index.js]: use [es6-promisify](https://github.com/digitaldesignlabs/es6-promisify) to simplify this.

File: index.js [hunk 4]
```diff
@@ -1,110 +1,139 @@
...
+    github.pullRequests.get(getMsgFormPrUrl(url), (err, res) => {
       if (err) {
-        console.error(err && err.stack);
-        return;
+        reject(err);
```
Comment (yorkie) [inline index.js]: `return` is required even though you called `reject` function.

File: index.js [hunk 5]
```diff
@@ -1,110 +1,139 @@
...
-        }, (err, res) => {
-          if (err) {
-            console.log(err && err.stack);
-            return;
-          }
-          console.log(res);
-        });
-      });
+function getPackageJsonFromHeadRepo(msg) {
+  return new Promise((resolve, reject) => {
```
Comment (yorkie) [inline index.js]: Ditto, use es6-promisify

File: index.js [hunk 6]
```diff
@@ -1,110 +1,139 @@
...
+    github.repos.getContent(msg, (err, res) => {
+      if (err) {
+        reject(err);
```
Comment (yorkie) [inline index.js]: Ditto, `return` is missing

File: github.js (hunk 1 above)
Comment (yorkie) [inline github.js]: Ok, that's fine, just leave this deletion

**Comment by yorkie** [comment]
@zhangweijie-cn please use es6/native Promise and es6-promisify to refine your code firstly, thanks :-0

File: index.js [hunk 7]
```diff
@@ -1,110 +1,107 @@
...
+  return {
+    user: list[3],
+    repo: list[4],
+    branch: list[6],
+  };
+}
+
+const getPullRequests = promisify(github.pullRequests.get);
+const getContent = promisify(github.repos.getContent);
+const updateFile = promisify(github.repos.updateFile);
```
Comment (yorkie) [inline index.js]: To be more safe, use `promisify(github.repos.updateFile.bind(github.repos))`. The same to the above 2 lines :-)

File: index.js [hunk 8]
```diff
@@ -1,110 +1,107 @@
...
+    };
+    const headMsg = {
+      user: res.head.repo.owner.login,
+      repo: res.head.repo.name,
+      path: 'package.json',
+      ref: res.head.ref,
+    };
+    return Promise.all([getContent(baseMsg), getContent(headMsg)]);
+  })
+  .then((baseRes, headRes) => {
```
Comment (yorkie) [inline index.js]: `then` function should always return only 1 argument, so you should destructor them inside function like this:

```js
then((results) => {
  const {baseRes, headRes} = results;
  // ...
})
```

File: index.js [hunk 9]
```diff
@@ -1,110 +1,107 @@
...
+    const basePackageJson = JSON.parse(new Buffer(baseRes.content, 'base64').toString());
+    const headPackageJson = JSON.parse(new Buffer(baseRes.content, 'base64').toString());
+    headPackageJson.version = bumpVersion(masks, basePackageJson.version);
+    const msg = getMsgFormHtmlUrl(headRes.html_url);
+    msg.path = 'package.json';
+    msg.message = 'bump version to ' + headPackageJson.version;
```
Comment (yorkie) [inline index.js]: suggest using message below:

```
version: bump to :version
```

File: index.js [hunk 10]
```diff
@@ -1,110 +1,116 @@
...
+  },
+}));
+
 app.use(githubWebhookHandler.middleware());
 app.listen((process.env.PORT || 4444));
+
+exports.getMasksFromCommentBody = getMasksFromCommentBody;
+exports.getMsgFormPrUrl = getMsgFormPrUrl;
+exports.getMsgFormHtmlUrl = getMsgFormHtmlUrl;
+exports.issueCommentHandler = issueCommentHandler;
```
Comment (yorkie) [inline index.js]: should move this to another file.

**Comment by zhangweijie-cn** [comment]
@yorkie can we merge first? I have test the server and I will write test case with sinonjs later

File: lib/function.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
+'use strict';
```
Comment (yorkie) [inline lib/function.js]: rename it "lib/utils.js"

File: lib/issueCommentHandler.js [hunk 12]
```diff
@@ -0,0 +1,60 @@
+'use strict';
```
Comment (yorkie) [inline lib/issueCommentHandler.js]: use "lib/issue-comment-handler.js"

File: lib/issueCommentHandler.js [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+      repo: res.base.repo.name,
+      path: 'package.json',
+      ref: res.base.ref,
+    };
+    const headMsg = {
+      user: res.head.repo.owner.login,
+      repo: res.head.repo.name,
+      path: 'package.json',
+      ref: res.head.ref,
+    };
```
Comment (yorkie) [inline lib/issueCommentHandler.js]: Suggest to use `try-catch` statement to wrap from 29-40 lines.

File: lib/issueCommentHandler.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  })
+  .then((results) => {
+    const baseRes = results[0];
+    const headRes = results[1];
+    const basePackageJson = JSON.parse(new Buffer(baseRes.content, 'base64').toString());
+    const headPackageJson = JSON.parse(new Buffer(headRes.content, 'base64').toString());
+    headPackageJson.version = bumpVersion(masks, basePackageJson.version);
+    const msg = getBranchMsgFormHtmlUrl(headRes.html_url);
+    msg.path = 'package.json';
+    msg.message = `version: ${basePackageJson.version} bump to :version ${headPackageJson.version}`;
```
Comment (yorkie) [inline lib/issueCommentHandler.js]: Should be:

```js
msg.message = `version: ${basePackageJson.version} bump to v${headPackageJson.version}`
```

**Comment by yorkie** [comment]
LGTM with few nits :-)

File: lib/issue-comment-handler.js [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+      };
+      const headMsg = {
+        user: res.head.repo.owner.login,
+        repo: res.head.repo.name,
+        path: 'package.json',
+        ref: res.head.ref,
+      };
+      return Promise.all([getContent(baseMsg), getContent(headMsg)]);
+    } catch (e) {
+      return Promise.reject(e);
```
Comment (yorkie) [inline lib/issue-comment-handler.js]: use `throw new Error('Github Invalid Request');`

**Comment by zhangweijie-cn** [comment]
@yorkie

File: lib/utils.js [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    MAJOR = 1;
+  } else if (commentBody.match('minor')) {
+    MINOR = 1;
+  } else if (commentBody.match('patch')) {
+    PATCH = 1;
+  } else {
+    return null;
+  }
+  const masks = [MAJOR, MINOR, PATCH];
+  return masks;
```
Comment (yorkie) [inline lib/utils.js]: return directly :-)

File: lib/utils.js [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+  };
+}
+
+function getBranchMsgFormHtmlUrl(url) {
+  const list = url.split('/');
+  return {
+    user: list[3],
+    repo: list[4],
+    branch: list[6],
+  };
```
Comment (yorkie) [inline lib/utils.js]:
```js
const m = /https:\/\/github\.com\/(\w+)\/(\w+)\/blob\/(\w+)\/package.json/;
return { user: m[1], repo: m[2], branch: m[3] };
```

**Comment by zhangweijie-cn** [comment]
@yorkie

**Pull request merged**